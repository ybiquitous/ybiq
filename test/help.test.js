import assert from 'assert'
import exec from './helpers/exec'

const HELP = `
cli.js <command>

Commands:
  cli.js init  Setup npm project:
               - Update 'package.json'
               - Create '.commitlintrc.js'
               - Create '.editorconfig'
               - Create '.eslintrc.js'

Options:
  --help, -h     Show help                                             [boolean]
  --version, -v  Show version number                                   [boolean]
`.trim()

suite('help', () => {
  ;[[], ['unknown'], ['unknown', 'xyz']].forEach(args => {
    test(`with arguments [${args.join(', ')}]`, async () => {
      const error = await exec(...args).catch(err => err)
      const { code, stdout, stderr } = error
      assert(error instanceof Error)
      assert(code === 1)
      assert(stdout === '')
      assert(stderr.includes(HELP), stderr)
    })
  })
  ;['--help', '-h'].forEach(option => {
    test(`with "${option}" option`, async () => {
      const { stdout, stderr } = await exec(option)
      assert(stdout.includes(HELP), stdout)
      assert(stderr === '')
    })
  })
})
