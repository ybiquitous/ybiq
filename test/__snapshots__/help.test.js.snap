// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`with "--help" option 1`] = `
"cli.js <command>

Commands:
  cli.js init  Setup npm project, e.g. update 'package.json'

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]
"
`;

exports[`with "-h" option 1`] = `
"cli.js <command>

Commands:
  cli.js init  Setup npm project, e.g. update 'package.json'

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]
"
`;

exports[`with arguments [] 1`] = `
"Command failed: ./lib/cli.js
cli.js <command>

Commands:
  cli.js init  Setup npm project, e.g. update 'package.json'

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`with arguments []: stderr 1`] = `
[Error: Command failed: ./lib/cli.js
cli.js <command>

Commands:
  cli.js init  Setup npm project, e.g. update 'package.json'

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]

Not enough non-option arguments: got 0, need at least 1
]
`;

exports[`with arguments [unknown, xyz] 1`] = `
"Command failed: ./lib/cli.js unknown xyz
cli.js <command>

Commands:
  cli.js init  Setup npm project, e.g. update 'package.json'

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]

Unknown arguments: unknown, xyz
"
`;

exports[`with arguments [unknown, xyz]: stderr 1`] = `
[Error: Command failed: ./lib/cli.js unknown xyz
cli.js <command>

Commands:
  cli.js init  Setup npm project, e.g. update 'package.json'

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]

Unknown arguments: unknown, xyz
]
`;

exports[`with arguments [unknown] 1`] = `
"Command failed: ./lib/cli.js unknown
cli.js <command>

Commands:
  cli.js init  Setup npm project, e.g. update 'package.json'

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]

Unknown argument: unknown
"
`;

exports[`with arguments [unknown]: stderr 1`] = `
[Error: Command failed: ./lib/cli.js unknown
cli.js <command>

Commands:
  cli.js init  Setup npm project, e.g. update 'package.json'

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]

Unknown argument: unknown
]
`;
